app/controllers/control_panel/control_panel_controller.rb

```diff
@@ -6,7 +6,16 @@ class ControlPanel::ControlPanelController < ApplicationController
 
   def require_user
     return if current_user
-    redirect_to root_path, alert: t('user.auth.failure')
+    redirect_to root_path, alert: t('auth.failure.user')
   end
 
+  def require_owner
+    return if current_user.id == find_board.owner_id
+    redirect_to control_panel_board_path(find_board), alert: t('auth.failure.owner')
+  end
+
+  def require_member
+    return if find_board.members.map { |a| a.user_id}.include?(current_user.id)
```

Comment:
You can't check it from other side ? I remember that you have has_many through association so:
`current_user.boards.find_by(id: find_board.id)`
In this way you can avoid iteration.